### PR TITLE
Expression evaluation autocomplete overlay is positioned over the last `.` in the expression

### PR DESCRIPTION
**Pull request by elliette**
- Previously we had a boolean parameter `tracking` which if `true` would display the autocomplete overlay over the cursor
- Instead, clients now provide a callback function that determines the X-position of the overlay. For the case of the expression evaluation autocomplete, this callback returns the position of the last `.` in the expression (if there is a `.`). 

**Demo:**

expression evaluation 2

This PR is part of the work towards https://github.com/flutter/devtools/issues/3095

### CODE REVIEW

File: packages/devtools_app/lib/src/ui/search.dart [hunk 1]
```diff
@@ -556,6 +556,11 @@ typedef ClearSearchField = Function(
   bool force,
 });
 
+/// Provided by clients to specify where the autocomplete overlay should be
+/// positioned relative to the input text.
+typedef DetermineOverlayXPosition = double Function(
+    String inputValue, TextStyle inputStyle);
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/ui/search.dart:562]: nit: trailing comma

File: packages/devtools_app/lib/src/debugger/evaluate.dart [hunk 2]
```diff
@@ -135,14 +136,27 @@ class _ExpressionEvalFieldState extends State<ExpressionEvalField>
...
                 border: OutlineInputBorder(),
                 focusedBorder: OutlineInputBorder(borderSide: BorderSide.none),
                 enabledBorder: OutlineInputBorder(borderSide: BorderSide.none),
                 labelText: 'Eval',
               ),
+              determineOverlayXPosition:
+                  (String inputValue, TextStyle inputStyle) {
+                // X-coordinate is equivalent to the width of the input text
+                // up to the last "." or the insertion point (cursor):
+                final textSegment = inputValue.contains('.')
```
Comment (jacob314) [inline packages/devtools_app/lib/src/debugger/evaluate.dart:150]: nit: slightly cleaner is:
```dart
final indexOfDot = inputValue.lastIndexOf('.')
final textSegment = indexOfDot != -1 ? inputValue.substring(0, indexOfDot + 1) : inputValue;
```
A little better as it doesn't have two different expressions with the same goal (contains and lastIndexOf) and is a little faster due to less dupe work not that it matters in this particular case.

**Review by jacob314 — APPROVED** [review]
lgtm

File: packages/devtools_app/lib/src/ui/search.dart [hunk 3]
```diff
@@ -556,6 +556,11 @@ typedef ClearSearchField = Function(
   bool force,
 });
 
+/// Provided by clients to specify where the autocomplete overlay should be
+/// positioned relative to the input text.
+typedef DetermineOverlayXPosition = double Function(
```
[the remark this answers is not in the document]
Comment (elliette) [reply packages/devtools_app/lib/src/ui/search.dart]: Done, renamed. Still using the typedef since it's being used twice

**Review by jacob314 — COMMENTED** [review]
lgtm

**Pull request merged**